### ISSUE TITLE
dmypy doesn't working when ignore_missing_imports in [tool.mypy] session

### ISSUE THREAD

**Issue by uf-fipper** [python/mypy#14272]
**Bug Report**

dmypy doesn't working when `ignore_missing_imports` in `[tool.mypy]` session.
I run the command on win11.

**To Reproduce**

> `pyproject.toml`
>
> ```toml
> [tool.mypy]
> ignore_missing_imports = true
> ```
>
> `dmypy run a.py`
>
> ```
> Restarting: configuration changed
> Daemon stopped
> Daemon started
> Response: {'restart': 'configuration changed', 'platform': 'win32', 'python_version': '3_9', 'roundtrip_time': 0.47419118881225586}
> ```

> `pyproject.toml`
>
> ```toml
> [[tool.mypy.overrides]]
> module = ["apscheduler"]
> ignore_missing_imports = true
> ```
>
> `dmypy run a.py`
>
> ```
> Success: no issues found in 1 source file
> ```

**Your Environment**

- Mypy version used: 0.991
- Mypy command-line flags: `dmypy run a.py`
- Mypy configuration options from `pyproject.toml`:

```toml
[tool.mypy]
ignore_missing_imports = true
```

- Python version used: 3.9
- os: **win11**

**Comment by ethanhs** [comment]
Probably the same issue as https://github.com/python/mypy/issues/14241

**Comment by jbaudisch** [comment]
I got the same issue when using VSCode's [Mypy Type Checker](https://marketplace.visualstudio.com/items?itemName=ms-python.mypy-type-checker). Having `ignore_missing_imports = true` in my pyproject.toml leads to a non-working extension (in terms of not showing any errors). If I comment this out, dmypy does work again (errors are shown again). Don't know if this issue still exist or if it is a problem in the extension. But if I switch to mypy instead of dmypy in the extension, it works - so I think the problem is here.
